src/messaging/ReliableMessageMgr.cpp

```diff
@@ -471,6 +465,27 @@ CHIP_ERROR ReliableMessageMgr::MapSendError(CHIP_ERROR error, uint16_t exchangeI
     return error;
 }
 
+void ReliableMessageMgr::CalculateNextRestranTime(RetransTableEntry * entry)
+{
+    System::Clock::Timestamp baseTimeout = System::Clock::Milliseconds64(0);
+
+    if (entry->ec->IsPeerActive())
+    {
+        // If we know the peer is active with the exchangeContext, use the Active Retrans timeout
+        baseTimeout = entry->ec->GetSessionHandle()->GetRemoteMRPConfig().mActiveRetransTimeout;
+    }
+    else
+    {
+
+        // If the exchange context doesn't know if the peer is active or not.
+        // Choose active/idle timeout from PeerActiveMode of session per 4.11.2.1. Retransmissions.
+        baseTimeout = entry->ec->GetSessionHandle()->GetMRPBaseTimeout();
```

Comment:
MRP base timeout is used in other places, such as data model timeout computation. Is this now no longer valid?

Reply:
Data model computes a "worst case" timeout, as I recall....  In particular, it does not use `GetMRPBaseTimeout()`, right?

Reply:
`GetMRPBaseTimeout()` only gets used in the two location where it was replaced with the `CalculateNextRestranTime(...)`